pallets/gear/src/benchmarking/syscalls.rs

```diff
@@ -280,12 +280,9 @@ where
         for i in 0..(API_BENCHMARK_BATCH_SIZE * repetitions) {
             // free them in steps
             let start = i as i32 * pages_per_call as i32;
-            instructions.extend([
-                I32Const(start),
-                I32Const(start + pages_per_call as i32 - 1),
-                Call(1),
-                I32Const(0),
-            ]);
+            let end = i32::max(start + pages_per_call as i32 - 1, 0);
```

Comment:
Eh, what the reason of i32::max(_, 0) using ? end can be negative?

Reply:
If you wanna to avoid overflow, then use checked_add(_).unwrap()

Reply:
0th repetition of 0 pages per call minus 1 can become negative, so free range will be called with 0, -1

Reply:
and -1 is required so when pages_per_call is 1, we call exactly free_range(page, page), since it's range is inclusive